### PR TITLE
Testing Tic Tac Toe with RSpec

### PR DESCRIPTION
**Pull request by SmithVyne**
- We used the RSpec tester to test the functions of our game.
- We used expectations to check for expected outputs.

_**Dear TSE, If you have a comment on our tests kindly please explain ALL required points you have for the project, Not only 1 or 2 of them, this will allow us to understand what you need better and do it ALL one time.  Thanks in advance for understanding.**_

### CODE REVIEW

**Comment by SmithVyne** [comment]
Thank you @rOluochKe for your review

**Comment by SmithVyne** [comment]
Thank you @rOluochKe for your review

**Comment by AhmedTohamy01** [comment]
Hi man @rOluochKe 

Is there is anything specific required , like required test scenarios for required point in the program ?

Your detailed answer will help us a lot

**Comment by rOluochKe** [comment]
> Hi man @rOluochKe
> 
> Is there is anything specific required , like required test scenarios for specific points in the program?
> 
> For example,
> it should have at least 10 tests ?
> it should have a test to check the player names or not ?
> 
> Your detailed answer will help us a lot

### Sure, you can have at least 10 test, things like checking names, if a player enter wrong input, the message, does it switch players

**Comment by SmithVyne** [comment]
All required changes have been added
**The only lint error is the one in the Rspec default files (spec/spec_helper.rb)**. We shouldn't edit that file so we've left it.
**Please approve the project and let's move forward if it's okay. :pray: **

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  # This option will default to `:apply_to_host_groups` in RSpec 4 (and will
+  # have no way to turn it off -- the option exists only for backwards
+  # compatibility in RSpec 3). It causes shared context metadata to be
+  # inherited by the metadata hash of host groups and examples, rather than
+  # triggering implicit auto-inclusion in groups with matching metadata.
+  config.shared_context_metadata_behavior = :apply_to_host_groups
+
+# The settings below are suggested to provide a good initial experience
+# with RSpec, but feel free to customize to your heart's content.
+=begin
```
Comment (kobiyoyo) [inline spec/spec_helper.rb:49]: You can use this a guide https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Style/BlockComments
````ruby
# bad
=begin
Multiple lines
of comments...
=end

# good
# Multiple lines
# of comments...
````

**Comment by AhmedTohamy01** [comment]
Dear TSE, 

Did you rejected the project and **_DELAYED_** our progress just because the comments in "spec_helper" file is written in a "bad way" ?

We are not the one who wrote the "spec_helper" by the way,  and we should NOT edit it. You can send email to David Chelimsky who built the "RSpec" to tell him this.

Why we don't get the approval if everything required is done?

**Comment by kobiyoyo** [comment]
Hello @AhmedTohamy01 ,hope you are doing great?
This project can't be approved until you fix the linter error. Secondly what i wrote is just a suggestion.
Do have a nice weekend

**Comment by AhmedTohamy01** [comment]
Dear Desmond, 

I think you didn't notice that the liner error is in "spec_helper" file also. This is a system file for the RSpec testing utility, and it's not our file. We can NOT change this.

**Comment by AhmedTohamy01** [comment]
While i don't understand why you want us to fix Lint Issues in files **_NOT belong to us_**, but it's okay. we fixed it now. Can we have the approval now ?

**Comment by Oluwadamilareolusakin** [comment]
Great job team! You may merge 🙂

**Comment by AhmedTohamy01** [comment]
This was really fast!

**_Thank you_**

**Pull request merged**